packages/smooth_app/lib/pages/preferences/user_preferences_faq.dart

```diff
@@ -281,6 +281,15 @@ class UserPreferencesFaq extends AbstractUserPreferences {
                         icon: Icons.open_in_new,
                       ),
                       const SizedBox(height: VERY_SMALL_SPACE),
+                      SmoothAlertContentButton(
+                        onPressed: () => LaunchUrlHelper.launchURL(
+                          'https://openfoodfacts.org/privacy',
+                          true,
+                        ),
```

Comment:
```suggestion
                        onPressed: () async => LaunchUrlHelper.launchURL(
                          UriHelper.replaceSubdomain(
                            Uri.parse(
                              'https://world.openfoodfacts.org/privacy',
                            ),
                          ),
                          true,
                        ),
```

Reply:
@monsieurtanuki 

I think `LaunchUrlHelper.launchURL` automatically does that if passed with true as Boolean, so in a way it gets localized. 

Can you plz have a look once

Reply:
@AshAman999 You're right!

Reply:
The string should be made translatable / adapt to the language

Reply:
@teolemon Actually in the code the URL is "localized", country-wise.
The `true` parameter is not very explicit, I'll give you that, and it only sets the country - not the language, e.g. https://be.openfoodfacts.org/privacy (and not https://be-fr.openfoodfacts.org/privacy or https://be.openfoodfacts.org/privacy)

Unfortunately there's some confusion between off-dart and Smoothie.

off-dart version, method `replaceSubdomainWithCodes` in `UriHelper`:
```dart
  /// Replaces the subdomain of an URI with specific country and language.
  ///
  /// No default language nor country: null means no parameter.
  /// For instance
  /// * https://world.xxx... would be standard
  /// * https://world-fr.xxx... would be "no country" in French
  /// * https://fr.xxx... would be France
  /// * https://fr-es.xxx... would be France in Spanish
  static Uri replaceSubdomainWithCodes(
    final Uri uri, {
    final String? languageCode,
    String? countryCode,
  }) {
```

Smoothie version, method `_replaceSubdomainWithCodes` in `LaunchUrlHelper`:
```dart
  static String _replaceSubdomainWithCodes(String url) {
    if (!url.contains('https://openfoodfacts.')) {
      throw 'Error do not use local identifier in url';
    }

    String? countryCode =
        PlatformDispatcher.instance.locale.countryCode?.toLowerCase();

    if (countryCode == null) {
      countryCode = 'world.';
    } else {
      countryCode = '$countryCode.';
    }

    url = url.replaceAll(
        'https://openfoodfacts.', 'https://${countryCode}openfoodfacts.');

    return url;
  }

  static Future<void> launchURL(String url, bool isOFF) async {
    if (isOFF) {
      url = _replaceSubdomainWithCodes(url);
    }
// ...
  }
```

@AshAman999 I guess @teolemon is right and we should use my code suggestion. And merge the PR.

Meanwhile, I'm about to open an issue about using slightly different `replaceSubdomainWithCodes` and `_replaceSubdomainWithCodes` methods.